src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/TypeCast.cs

```diff
@@ -432,7 +431,7 @@ private static unsafe object CheckCastClassSpecial(MethodTable* pTargetType, obj
                     goto done;
 
                 // They don't cast to any other class
-                goto fail;
+                ThrowHelper.ThrowInvalidCastException();
```

Comment:
Is this introducing a frame to this method? This structuring was obviously deliberate and matches `ChkCastClassSpecial` in the other corelib. What motivates this change?